### PR TITLE
ddl: grant nonexistent user/role incompatible error

### LINKED ISSUE #23649: grant nonexistent user/role incompatible error
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

I am logged in as root with `SUPER` privilege. The MySQL error messages returned are more helpful than the TiDB ones:

```sql
DROP USER IF EXISTS u1;
CREATE USER u1;
GRANT bogusrole to u1;
GRANT bogusrole to nonexisting;
```

### 2. What did you expect to see? (Required)

```sql
mysql [localhost:8023] {root} ((none)) > GRANT bogusrole to u1;
ERROR 3523 (HY000): Unknown authorization ID `bogusrole`@`%`
mysql [localhost:8023] {root} ((none)) > GRANT bogusrole to nonexisting;
ERROR 3523 (HY000): Unknown authorization ID `nonexisting`@`%`

```

### 3. What did you see instead (Required)

```sql
mysql> GRANT bogusrole to u1;
ERROR 1396 (HY000): Operation GRANT ROLE failed for `bogusrole`@`%`
mysql> GRANT bogusrole to nonexisting;
ERROR 1396 (HY000): Operation GRANT ROLE failed for `bogusrole`@`%`
```

### 4. What is your TiDB version? (Required)

```sql
mysql> SELECT tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v4.0.0-beta.2-2504-gb1b58d94e-dirty
Edition: Community
Git Commit Hash: b1b58d94e706183ebb3447126491e94bdfd04cfc
Git Branch: initial-support-dynamic-privs
UTC Build Time: 2021-03-29 20:50:19
GoVersion: go1.13.15
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by Howie59**
### What problem does this PR solve?

Issue Number: close #23649 

### Check List 

Tests 

- Unit test

### Release note 

- No release note

### CODE REVIEW

File: executor/simple_test.go [hunk 1]
```diff
@@ -871,3 +871,15 @@ func (s *testSuite3) TestIssue17247(c *C) {
...
+// Close issue #23649.
+// See https://github.com/pingcap/tidb/issues/23649
+func (s *testSuite3) TestIssue23649(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("DROP USER IF EXISTS issue23649;")
+	tk.MustExec("CREATE USER issue23649;")
+	_, err := tk.Exec("GRANT bogusrole to issue23649;")
+	c.Assert(err.Error(), Equals, "[executor:3523]Unknown authorization ID `bogusrole`@`%`")
+	_, err = tk.Exec("GRANT bogusrole to nonexisting;")
+	c.Assert(err.Error(), Equals, "[executor:3523]Unknown authorization ID `bogusrole`@`%`")
```
Comment (xiongjiwei) [inline executor/simple_test.go:884]: well, this test result is different from the issue. I have not verified it by myself, please check it again.

Comment (Howie59) [reply executor/simple_test.go:884]: i execute in mysql, and get **ERROR 3523 (HY000): Unknown authorization ID `bogusrole`@`%`** twice

Comment (xiongjiwei) [reply executor/simple_test.go:884]: great, thank you

**Comment by xiongjiwei** [comment]
@Howie59 hi, there is a private repo that should be changed, you may need run test like `/run-all-tests tidb-test=pr/1180`

**Comment by hi-rustin** [comment]
@Howie59 Could you please take a look?

It seems the CI failed cause by your changes.

```sh
[2021-04-15T03:02:40.718Z] time="2021-04-15T11:02:40+08:00" level=fatal msg="run test [role] err: sql:GRANT sys_role TO peter@clickhost.net;: failed to run query \n\"GRANT sys_role TO peter@clickhost.net;\" \n around line 41, \nwe need(91):\nGRANT sys_role TO peter@clickhost.net;\nError 1396: Operation GRANT ROLE failed for `sys_rol\nbut got(91):\nGRANT sys_role TO peter@clickhost.net;\nError 3523: Unknown authorization ID `sys_role`@`%`\n\n"

[2021-04-15T03:02:40.718Z] + echo 'tidb-server(PID: 19923) stopped'

[2021-04-15T03:02:40.718Z] tidb-server(PID: 19923) stopped

[2021-04-15T03:02:40.718Z] + kill -9 19923

```

 log: https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_common_test/detail/tidb_ghpr_common_test/37218/pipeline

**Comment by morgo** [comment]
There is a bit of a backlog of PRs merging right now. Since this requires the tidb-test to also be merged, I will wait for the backlog to clear, and merge this if it is still open when I start work tomorrow (~12 hours time)

**Comment by morgo** [comment]
It looks like the tidb-test fixes are non complete. After merging them I am still getting failures.

**Comment by xiongjiwei** [comment]
@morgo I think pr/1183 is a complete fix, it is totally the same with pr/1180.

**Pull request merged**